hydra-node/src/Hydra/Node.hs

```diff
@@ -89,8 +89,8 @@ data HydraNode tx m = HydraNode
   }
 
 data HydraNodeLog tx
-  = BeginEvent {by :: Party, event :: Event tx}
-  | EndEvent {by :: Party, event :: Event tx}
+  = BeginEvent {by :: Party, eventId :: Word64, event :: Event tx}
+  | EndEvent {by :: Party, eventId :: Word64}
```

Comment:
Why not `Natural`? Word64 will be more than big enough, but we used `Natural` in the past for non-negative numbers

Should: use Natural unless there is a specific reason for Word64

Reply:
`Word64` is slightly simpler than `Natural` esp. when it comes to generating values: It's supported by QC out of the box whereas Natural is not.

Reply:
If you care, I can change it.

Reply:
In another PR.

Reply:
Okay, I get that argument. We should migrate all `Natural` in the repo then to `Word64` as we pass it.